### PR TITLE
chore: Allow newer Node versions

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Updates to:
- Web's Makefile to allow newer versions of Node (17.x, 18.x, 19.x)
- `.nvmrc` for CI to test in newer Node
- Documentation and package.json

## Test Plan

Passes tests and builds WebUI successfully

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/Makefile [hunk 1]
```diff
@@ -64,7 +64,7 @@ check-package-lock:
 	if grep 'ssh://' package-lock.json ; then echo "ssh url in package-lock.json, please convert to https url" ; false ; fi
 .PHONY: check-requirements
 check-requirements:
-	node --version | grep -E 'v16\.(1[3-9]|2[0-9])' || (echo "node version >=16.13 <17 is required" ; false)
+	node --version | grep -E 'v16\.(1[3-9]|2[0-9])|v17|v18|v19' || (echo "node version >=16.13 <20 is required" ; false)
```
Comment (hkang1) [inline webui/react/Makefile:67]: I think that long syntax was just a fancy way of saying newer than node 16.13 and up to 16.29 (inclusive). So we needed 16.13+ at some point but I'm not sure if we still do at this point. Up to you if you want to bump the version requirements up even higher.

**Review by hkang1 — APPROVED** [review]
LGTM!

File: webui/react/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mapmeld) [reply webui/react/Makefile]: ✅

Comment (mapmeld) [reply webui/react/Makefile]: In Slack, Keita and I agreed to just be permissive now and revisit requirements when v20 (the next long-term support version) is available later this year

**Pull request merged**